samples/nrf9160/ciphersuites/src/main.c

```diff
@@ -0,0 +1,250 @@
+		return err;
+	}
+
+	/* Associate the socket with the security tag
+	 * we have provisioned the certificate with.
+	 */
+	err = setsockopt(fd, SOL_TLS, TLS_SEC_TAG_LIST, tls_sec_tag,
+			 sizeof(tls_sec_tag));
+	if (err) {
+		printk("Failed to setup TLS sec tag, err %d, %s\n", errno, strerror(errno));
+		return err;
+	}
+
+	err = setsockopt(fd, SOL_TLS, TLS_HOSTNAME, HTTPS_HOSTNAME, sizeof(HTTPS_HOSTNAME) - 1);
+	if (err) {
+		printk("Failed to setup TLS hostname, err %d, %s\n", errno, strerror(errno));
+		return err;
+	}
+	return 0;
+}
+
+void main(void)
+{
+	int err;
+	int fd;
+	struct addrinfo *res;
+	struct addrinfo hints = {
+		.ai_family = AF_INET,
+		.ai_socktype = SOCK_STREAM,
+	};
+	nrf_sec_cipher_t ciphersuite_list[] = { 0 };
+
+	printk("Ciphersuites sample started\n\r");
+
+#if !defined(CONFIG_SAMPLE_TFM_MBEDTLS)
+	/* Provision certificates before connecting to the LTE network */
+	err = cert_provision();
+	if (err) {
+		return;
+	}
+#endif
+
+	printk("Waiting for network.. ");
+	err = lte_lc_init_and_connect();
+	if (err) {
+		printk("Failed to connect to the LTE network, err %d\n", err);
+		return;
+	}
+	printk("OK\n");
+
+	err = getaddrinfo(HTTPS_HOSTNAME, NULL, &hints, &res);
+	if (err) {
+		printk("getaddrinfo() failed, err %d, %s\n", errno, strerror(errno));
+		return;
+	}
+
+	((struct sockaddr_in *)res->ai_addr)->sin_port = htons(HTTPS_PORT);
+
+	printk("Trying all ciphersuites to find which ones are supported...\n");
+
+	for (int i = 0; i < sizeof(ciphersuites) / sizeof(struct lookup_ciphersuite); i++) {
+		if (IS_ENABLED(CONFIG_SAMPLE_TFM_MBEDTLS)) {
+			fd = socket(AF_INET, SOCK_STREAM | SOCK_NATIVE_TLS, IPPROTO_TLS_1_2);
+		} else {
+			fd = socket(AF_INET, SOCK_STREAM, IPPROTO_TLS_1_2);
+		}
+		if (fd == -1) {
+			printk("Failed to open socket!\n");
+			goto clean_up;
+		}
+
+		/* Setup TLS socket options */
+		err = tls_setup(fd);
+		if (err) {
+			goto clean_up;
+		}
+
+		printk("Trying ciphersuite: %s\n", ciphersuites[i].name);
+
+		ciphersuite_list[0] = ciphersuites[i].value;
+
+		err = setsockopt(fd, SOL_TLS, TLS_CIPHERSUITE_LIST,
+						 ciphersuite_list, sizeof(ciphersuite_list));
+		if (err) {
+			printk("Failed to setup ciphersuite list, err %d, %s\n",
+			       errno,
+			       strerror(errno));
+			goto clean_up;
+		}
+
+		printk("Connecting to %s... ", HTTPS_HOSTNAME);
+		err = connect(fd, res->ai_addr, sizeof(struct sockaddr_in));
+		if (err) {
+			printk("connect() failed, err: %d, %s\n", errno, strerror(errno));
+			ciphersuites[i].supported = false;
+		} else {
+			printk("Connected.\n");
+			ciphersuites[i].supported = true;
+		}
+
+		close(fd);
+	}
+
+	printk("\nCiphersuite support summary for host `%s`:\n", HTTPS_HOSTNAME);
+
+	for (int i = 0; i < sizeof(ciphersuites) / sizeof(struct lookup_ciphersuite); i++) {
+		printk("%s: %s\n", ciphersuites[i].name,
+		       ciphersuites[i].supported ? "Supported" : "No");
+	}
+
+	printk("\nFinished.\n");
+
+clean_up:
+	freeaddrinfo(res);
+	(void)close(fd);
```

Comment:
After the final iteration we'll try to close an already closed socket here. Would be better to set `fd = -1` at the end of the loop.